app/src/main/java/org/akvo/flow/util/SurveyFileNameGenerator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.util;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+public class SurveyFileNameGenerator {
+
+    public SurveyFileNameGenerator() {
+    }
+
+    @NonNull
+    public String generateFileName(@Nullable String entryName) {
+        if (entryName == null || entryName.isEmpty()) {
+            return "";
+        }
+        int fileSeparatorPosition = entryName.lastIndexOf("/");
```

Comment:
Ok so I have tried and it does not work. We need to use the filename which is the last bit of '/'. This is because when the bootstrap is generated even if the user created it as a name it is actually made a folder. We should maybe forbid some characters in the actual form name in the backend.